src/term/net_to_term.rs

```diff
@@ -131,28 +131,31 @@ pub fn net_to_term_non_linear(net: &INet, book: &Book) -> (Term, bool) {
           let arg_port = net.enter_port(Port(node, 1));
           let (arg_term, fst_valid) = reader(net, arg_port, namegen, dup_scope, tup_scope, book);
           let valid = op_valid && fst_valid;
-          match op_term {
-            Term::Num { val } => {
-              let (val, op) = split_num_with_op(val);
-              if let Some(op) = op {
-                // This is Num + Op in the same value
-                (Term::Opx { op, fst: Box::new(Term::Num { val }), snd: Box::new(arg_term) }, valid)
-              } else {
-                // This is just Op as value
-                (
+
+          fn go(op_term: Term, arg_term: Term) -> Term {
+            match op_term {
+              Term::Num { val } => {
+                let (val, op) = split_num_with_op(val);
+                if let Some(op) = op {
+                  // This is Num + Op in the same value
+                  Term::Opx { op, fst: Box::new(Term::Num { val }), snd: Box::new(arg_term) }
+                } else {
+                  // This is just Op as value
                   Term::Opx {
                     op: Op::from_hvmc_label(val).unwrap(),
                     fst: Box::new(arg_term),
                     snd: Box::new(Term::Era),
-                  },
-                  valid,
-                )
+                  }
+                }
               }
+              Term::Opx { op, fst, snd } => match &*snd {
+                Term::Era => Term::Opx { op, fst, snd: Box::new(arg_term) },
```

Comment:
Add a comment explaining that an Era means this came from the readback of the first OP2 node while anything else means a partially applied chain of OP2 nodes